### PR TITLE
add compute_min_max to AttrRangeFacetFilter

### PR DESCRIPTION
**Pull request by efremov-dma**

### CODE REVIEW

File: elasticmagic_qf_attrs/facet.py [hunk 1]
```diff
@@ -38,6 +39,54 @@
 return attrIds;
 '''
 
+RANGE_ATTR_MINMAX_MAP_SCRIPT = '''
+for (v in doc[params.field].values) {
+    String attr_id = (v >>> 32).toString();
+    double float_val = Float.intBitsToFloat((int) v);
```
Comment (anti-social) [inline elasticmagic_qf_attrs/facet.py:45]: Can be `float` instead of `double`

File: elasticmagic_qf_attrs/facet.py [hunk 2]
```diff
@@ -38,6 +39,54 @@
...
+    if (!state.containsKey(attr_id)) {
+        state[attr_id] = new HashMap();
```
Comment (anti-social) [inline elasticmagic_qf_attrs/facet.py:47]: We can use array with 2 elements:
```
float[] min_max = state.get(attr_id);
if (min_max == null) {
    state[attr_id] = new float[] {float_val, float_val};
    continue;
}
```

File: elasticmagic_qf_attrs/facet.py [hunk 3]
```diff
@@ -335,6 +420,26 @@ def _apply_agg(self, search_query: SearchQuery) -> SearchQuery:
...
             )
 
+        if self._compute_min_max:
+            min_max_agg = agg.ScriptedMetric(
+                map_script=RANGE_ATTR_MINMAX_MAP_SCRIPT,
+                reduce_script=RANGE_ATTR_MINMAX_REDUCE_SCRIPT,
+                params={
+                    'field': self.field,
+                },
+            )
```
Comment (anti-social) [inline elasticmagic_qf_attrs/facet.py:430]: Elasticsearch 7.x requires `combine_script` parameter

**Review by anti-social — CHANGES_REQUESTED** [review]

File: elasticmagic_qf_attrs/facet.py [hunk 4]
```diff
@@ -38,6 +39,54 @@
 return attrIds;
 '''
 
+RANGE_ATTR_MINMAX_MAP_SCRIPT = '''
+for (v in doc[params.field].values) {
```
[the remark this answers is not in the document]
Comment (efremov-dma) [reply elasticmagic_qf_attrs/facet.py]: fixed

File: elasticmagic_qf_attrs/facet.py (hunk 1 above)
Comment (efremov-dma) [reply elasticmagic_qf_attrs/facet.py]: fixed

File: elasticmagic_qf_attrs/facet.py (hunk 2 above)
Comment (efremov-dma) [reply elasticmagic_qf_attrs/facet.py]: fixed

File: elasticmagic_qf_attrs/facet.py [hunk 5]
```diff
@@ -38,6 +39,54 @@
...
+'''
+
+RANGE_ATTR_MINMAX_REDUCE_SCRIPT = '''
+Map reduced = new HashMap();
+for (state in states) {
+    for (entry in state.entrySet()) {
+        String attr_id = entry.getKey();
+        Map values = entry.getValue();
+        if (!reduced.containsKey(attr_id)) {
+            reduced[attr_id] = new HashMap();
```
[the remark this answers is not in the document]
Comment (efremov-dma) [reply elasticmagic_qf_attrs/facet.py]: fixed

File: elasticmagic_qf_attrs/facet.py (hunk 3 above)
Comment (efremov-dma) [reply elasticmagic_qf_attrs/facet.py]: fixed

**Pull request merged**